From akpm@linux-foundation.org Mon Jul  9 12:04:21 2007
From: Yinghai Lu <Yinghai.Lu@Sun.COM>
Date: Mon, 09 Jul 2007 12:03:09 -0700
Subject: USB: make the usb_device numa_node get assigned from controller
To: greg@kroah.com
Cc: linux-usb-devel@lists.sourceforge.net, akpm@linux-foundation.org, Yinghai.Lu@Sun.COM, ak@suse.de, clameter@sgi.com, rientjes@google.com, yinghai.lu@Sun.COM
Message-ID: <200707091903.l69J39cf002790@imap1.linux-foundation.org>


From: Yinghai Lu <Yinghai.Lu@Sun.COM>

So we can use dev_to_node(&usb_dev->dev) later in kmalloc_node to dma buffer

Signed-off-by: Yinghai Lu <yinghai.lu@sun.com>
Cc: Andi Kleen <ak@suse.de>
Acked-by: Christoph Lameter <clameter@sgi.com>
Cc: David Rientjes <rientjes@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/usb.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/core/usb.c
+++ b/drivers/usb/core/usb.c
@@ -253,6 +253,7 @@ usb_alloc_dev(struct usb_device *parent,
 	dev->dev.bus = &usb_bus_type;
 	dev->dev.type = &usb_device_type;
 	dev->dev.dma_mask = bus->controller->dma_mask;
+	set_dev_node(&dev->dev, dev_to_node(bus->controller));
 	dev->state = USB_STATE_ATTACHED;
 
 	INIT_LIST_HEAD(&dev->ep0.urb_list);
